src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -134,23 +134,33 @@ def get_attrs(
                     ).get("uuid")
             alert_rule_triggers.append(serialized)
 
-        alert_rule_projects = set()
-        for alert_rule in alert_rules.values():
+        # Query for all alert rules once more to get the latest 10 activations
+        # last_10_activations_subquery = AlertRuleActivations.objects.filter(
+        #     alert_rule_id=OuterRef("id")
+        # ).order_by("-date_added")[:10]
+
+        # # Filter alerts by id and annotate with the last 10 activations
+        # alert_rules_with_activations = AlertRule.objects.filter(id__in=alert_rules.keys()).annotate(
+        #     last_activations=list(last_10_activations_subquery)
+        # )
+        # for alert_rule in alert_rules_with_activations:
+        #     result[alert_rule]["activations"] = serialize(alert_rule.last_activations, **kwargs)
+
+        # Construct list of all projects associated with each alert rule
+        # given both the project fk as well as the snuba query projects relationship
+        projects_per_alert_rule = defaultdict(set)
```

Comment:
semi - unrelated to the main content in this change.

taking the opportunity to clean up this confusing logic.
we keep doing this confusing referencing stuff
```
rule_result = result[alert_rules[alert_rule_id]].setdefault("projects", [])
rule_result.append(project_slug)
```

instead, i'm now simply iterating through our item_list to build our project slug list.
No need to query AlertRule's again